File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] arr) {
        int n = arr.length;

        int  i=0,j=0,k=n-1;

        while(j<=k)
        {
            if(arr[j]==1)
            {
                j++;
            }
            else if(arr[j]==0)
            {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;j++;
            }
            else if(arr[j]==2)
            {
                int temp = arr[k];
                arr[k] = arr[j];
                arr[j] = temp;
                k--;
            }
        }


    }
}

File: 75-sort-colors/sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n = nums.size();
        int i = 0;
        int j = n-1;
        int k = 0;

        while(k<n)
        {
            while(i<=k && nums[k]==0)
            {
                swap(nums[i],nums[k]);
                i++;
            }
            while(k<=j && nums[k]==2)
            {
                swap(nums[j],nums[k]);
                j--;
            }
            while(i<=k && nums[k]==0)
            {
                swap(nums[i],nums[k]);
                i++;
            }
            k++;
           
        }
    }
};